crates/subspace-service/src/mmr/sync.rs

```diff
@@ -31,43 +122,17 @@ pub async fn mmr_sync<Block, Client, NR, OS>(
 where
     Block: BlockT,
     NR: NetworkRequest,
-    Client: HeaderBackend<Block>,
+    Client: ProvideRuntimeApi<Block> + HeaderBackend<Block>,
+    Client::Api: MmrApi<Block, H256, NumberFor<Block>>,
     OS: OffchainStorage,
 {
     debug!("MMR sync started.");
     let info = client.info();
     let protocol_name = generate_protocol_name(info.genesis_hash, fork_id.as_deref());
 
-    let mut offchain_db = OffchainDb::new(offchain_storage);
-
-    // Look for existing local MMR-nodes
-    let mut starting_position = {
-        let mut starting_position: Option<u32> = None;
-        for position in 0..=u32::MAX {
-            let canon_key = get_offchain_key(position.into());
-            if offchain_db
-                .local_storage_get(StorageKind::PERSISTENT, &canon_key)
-                .is_none()
-            {
-                starting_position = Some(position);
-                break;
-            }
-        }
-
-        match starting_position {
-            None => {
-                error!("Can't get starting MMR position - MMR storage is corrupted.");
-                return Err(sp_blockchain::Error::Application(
-                    "Can't get starting MMR position - MMR storage is corrupted.".into(),
-                ));
-            }
-            Some(last_processed_position) => {
-                debug!("MMR-sync last processed position: {last_processed_position}");
-
-                last_processed_position
-            }
-        }
-    };
+    let mut mmr = MmrOf::new(0, OffchainMmrStorage::new(offchain_storage));
+    let mut leaves_number = 0u32;
+    let mut starting_position = 0;
```

Comment:
Previously our first starting position was taken from our storage, now it’s always zero. Is that change intentional? 
It seems like it might slow down MMR sync as the chain gets longer.